src/rpc/blockchain.cpp

```diff
@@ -2219,7 +2225,10 @@ static RPCHelpMan savemempool()
         throw JSONRPCError(RPC_MISC_ERROR, "Unable to dump mempool to disk");
     }
 
-    return NullUniValue;
+    UniValue ret(UniValue::VOBJ);
+    ret.pushKV("filename", fs::PathToString(node.args->GetDataDirNet() / "mempool.dat"));
```

Comment:
In commit "Add filename to savemempool RPC result" (09c3395bc5efc0a969548e1d694a75458058342c)

Should replace `fs::PathToString(node.args->GetDataDirNet() / "mempool.dat")` with `(node.args->GetDataDirNet() / "mempool.dat").u8string()`. The `PathToString` and `u8string` functions do the same thing right now, but u8string is meant to be used for strings which need to be UTF8 (required to produce valid JSON), and PathToString is meant for internal use and can return paths which are not UTF-8.